src/daft-sql/src/modules/hashing.rs

```diff
@@ -0,0 +1,111 @@
+use daft_dsl::ExprRef;
+use daft_functions::{
+    hash::hash,
+    minhash::{minhash, MinHashFunction},
+};
+use sqlparser::ast::FunctionArg;
+
+use super::SQLModule;
+use crate::{
+    error::{PlannerError, SQLPlannerResult},
+    functions::{SQLFunction, SQLFunctionArguments, SQLFunctions},
+    unsupported_sql_err,
+};
+
+pub struct SQLModuleHashing;
+
+impl SQLModule for SQLModuleHashing {
+    fn register(parent: &mut SQLFunctions) {
+        parent.add_fn("hash", SQLHash);
+        parent.add_fn("minhash", SQLMinhash);
+    }
+}
+
+pub struct SQLHash;
+
+impl SQLFunction for SQLHash {
+    fn to_expr(
+        &self,
+        inputs: &[FunctionArg],
+        planner: &crate::planner::SQLPlanner,
+    ) -> SQLPlannerResult<ExprRef> {
+        match inputs {
+            [input] => {
+                let input = planner.plan_function_arg(input)?;
+                Ok(hash(input, None))
+            }
+            [input, seed] => {
+                let input = planner.plan_function_arg(input)?;
+                match seed {
+                    FunctionArg::Named { name, arg, .. } if name.value == "seed" => {
+                        let seed = planner.try_unwrap_function_arg_expr(arg)?;
+                        Ok(hash(input, Some(seed)))
+                    }
+                    arg @ FunctionArg::Unnamed(_) => {
+                        let seed = planner.plan_function_arg(arg)?;
+                        Ok(hash(input, Some(seed)))
+                    }
+                    _ => unsupported_sql_err!("Invalid arguments for hash: '{inputs:?}'"),
+                }
+            }
+            _ => unsupported_sql_err!("Invalid arguments for hash: '{inputs:?}'"),
+        }
+    }
+}
+
+pub struct SQLMinhash;
+
+impl TryFrom<SQLFunctionArguments> for MinHashFunction {
+    type Error = PlannerError;
+
+    fn try_from(args: SQLFunctionArguments) -> Result<Self, Self::Error> {
+        let num_hashes = args
+            .get_named("num_hashes")
+            .ok_or_else(|| PlannerError::invalid_operation("num_hashes is required"))?
+            .as_literal()
+            .and_then(|lit| lit.as_i64())
+            .ok_or_else(|| PlannerError::invalid_operation("num_hashes must be an integer"))?
+            as usize;
```

Comment:
How come we're transforming the literal to an `i64` and then casting it to `usize`?

Reply:
We're not transforming it to an i64 literal, we're extracting the i64 out of the literal. When we parse the sql ast all numerical types come through as either an i64 or f64. https://github.com/universalmind303/Daft/blob/46c5a7e39399dd08dbf7d96d22673cc35caf9a4b/src/daft-sql/src/planner.rs#L503.

Reply:
Oh okay, I see. Thanks!